vars/promoteRepos.groovy

```diff
@@ -0,0 +1,196 @@
+            withAWS(role: "${ARTIFACT_PROMOTION_ROLE_NAME}", roleAccount: "${AWS_ACCOUNT_ARTIFACT}", duration: 900, roleSessionName: 'jenkins-session') {
+                println("Pulling Prod ${repoType}")
+                sh("aws s3 sync s3://${ARTIFACT_PRODUCTION_BUCKET_NAME}/${RepoProdPath}/ ${artifactPath}/ --no-progress")
+            }
+
+        sh """
+            set -e
+            set +x
+
+            echo "Pulling ${revision} ${distType}"
+            cd ${artifactPath}
+            curl -SLO ${stagingPkgPathX64}
+            curl -SLO ${stagingPkgPathARM64}
+
+            ls -l
+        """
+
+        if (repoType.equals("yum")) {
+
+            println("Yum Repo Starts")
+
+            sh """
+                set -e
+                set +x
+
+                cd ${artifactPath}
+                rm -vf repodata/repomd.xml.asc
+    
+                echo "Update repo metadata"
+                createrepo --update .
+    
+                # Rename .xml to .pom for signing
+                # Please do not add .xml to signer filter
+                # As maven have many .xml and we do not want to sign them
+                # This is an outlier case for yum repo only
+                mv -v repodata/repomd.xml repodata/repomd.pom
+    
+                echo "Complete metadata update, awaiting signing repomd.xml"
+                cd -
+            """
+
+            signArtifacts(
+                artifactPath: "${artifactPath}/repodata/repomd.pom",
+                sigtype: '.sig',
+                platform: 'linux'
+            )
+
+            sh """
+                set -e
+                set +x
+    
+                cd ${artifactPath}/repodata/
+    
+                ls -l
+    
+                mv -v repomd.pom repomd.xml
+                mv -v repomd.pom.sig repomd.xml.sig
+    
+                # This step is required as yum only accept .asc and signing workflow only support .sig
+                cat repomd.xml.sig | gpg --enarmor | sed 's@ARMORED FILE@SIGNATURE@g' > repomd.xml.asc
+    
+                rm -vf repomd.xml.sig
+    
+                ls -l
+    
+                cd -
+            """
+        }
+
+        if (repoType.equals("apt")) {
+
+            println("Apt Repo Starts")
+
+            withCredentials([
+            string(credentialsId: 'jenkins-rpm-signing-account-number', variable: 'RPM_SIGNING_ACCOUNT_NUMBER'),
+            string(credentialsId: 'jenkins-rpm-signing-passphrase-secrets-arn', variable: 'RPM_SIGNING_PASSPHRASE_SECRETS_ARN'),
+            string(credentialsId: 'jenkins-rpm-signing-secret-key-secrets-arn', variable: 'RPM_SIGNING_SECRET_KEY_ID_SECRETS_ARN'),
+            string(credentialsId: 'jenkins-rpm-signing-key-id', variable: 'RPM_SIGNING_KEY_ID')]) {
+                echo 'APT Sign Repo'
+
+                withAWS(role: 'jenkins-prod-rpm-signing-assume-role', roleAccount: "${RPM_SIGNING_ACCOUNT_NUMBER}", duration: 900, roleSessionName: 'jenkins-signing-session') {
```

Comment:
Alot is happening within this credentials block. Just retrieve the credentials where required rather than doing the entire checks and stuff within this block.